packages/nodejs-scripts/src/endSemSummary.ts

```diff
@@ -90,12 +90,23 @@ GROUP BY userId,quizId`,
         percentage: (score / MAX_POINTS[quizId]) * 100,
       };
     }
-    for (const userData of Object.values(USER_ID_TO_QUIZ_SCORES)) {
+
+    for (const [userId, userData] of Object.entries(USER_ID_TO_QUIZ_SCORES)) {
       const quizPercentages = Object.values(userData.perQuiz).map(
         (quizData) => quizData.percentage
       );
       quizPercentages.sort((a, b) => b - a);
+
+      const excusedQuizzes = Object.entries(
+        TO_BE_EXCUSED[process.env.QUIZ_SUMMARY_COURSE_ID] || {}
+      ).filter(([_, userIds]) => (userIds as string[]).includes(userId)).length;
+
+      const remainingQuizzesAfterExcuses = TOP_N - excusedQuizzes;
+      const scalingFactor = remainingQuizzesAfterExcuses / TOP_N;
```

Comment:
Since `userData.sumTopN` is average of the top N quizzes, scaling is not neeeded.